p2p/simulations/adapters/exec.go

```diff
@@ -392,28 +406,90 @@ func execP2PNode() {
 	// them in a snapshot service
 	services := make(map[string]node.Service, len(serviceNames))
 	for _, name := range serviceNames {
-		serviceFunc, exists := serviceFuncs[name]
-		if !exists {
-			log.Crit("unknown node service", "name", name)
-		}
-		constructor := func(nodeCtx *node.ServiceContext) (node.Service, error) {
-			ctx := &ServiceContext{
-				RPCDialer:   &wsRPCDialer{addrs: conf.PeerAddrs},
-				NodeContext: nodeCtx,
-				Config:      conf.Node,
+		if name == "discovery" {
+			serviceFunc := func(ctx *ServiceContext) (node.Service, error) {
+				//addr := network.NewAddrFromNodeID(ctx.Config.ID)
```

Comment:
Ideally we should define a function which takes, not just the `Node ID` but also its `Underlay address`, rather than hardcoding it to 127.0.0.1:30303. I want to do that through `ctx.Config` but for some reason fields are not propagated correctly via the `ServiceContext`, so for now I have it hardcoded to `prt16`, which I get from scope.